src/transformations/extent.jl

```diff
@@ -6,19 +6,24 @@ calculating and adding an `Extents.Extent` to all objects.
 
 This can improve performance when extents need to be checked multiple times.
 """
-embed_extent(x; kw...) = apply(AbstractTrait, x; kw...)
+embed_extent(x; kw...) = apply(extent_applicator, GI.AbstractTrait, x; kw...)
 
+# We recursively run `embed_extent` so the lowest level
+# extent is calculated first and bubbles back up.
+# This means we touch each point only once.
 extent_applicator(x) = extent_applicator(trait(x), x)
 extent_applicator(::Nothing, xs::AbstractArray) = embed_extent.(xs)
-extent_applicator(::Union{AbstractCurveTrait,MultiPointTrait}, point) = point
-    
-function extent_applicator(trait::AbstractGeometryTrait, geom)
+function extent_applicator(trait::GI.AbstractGeometryTrait, geom)
     children_with_extents = map(GI.getgeom(geom)) do g
         embed_extent(g)
     end
     wrapper_type = GI.geointerface_geomtype(trait)
     extent = GI.extent(wrapper_type(children_with_extents))
-    return wrapper_type(children_with_extents, extent)
+    return wrapper_type(children_with_extents; extent, crs=GI.crs(geom))
 end
-extent_applicator(::PointTrait, point) = point
-extent_applicator(::PointTrait, point) = point
+function extent_applicator(trait::Union{GI.AbstractCurveTrait,GI.MultiPointTrait}, geom)
```

Comment:
What happens with points? I guess it doesn't make a lot of sense for someone to call this on a point, but if they did, does it make sense to have something specifically defined for that?

Reply:
Its just moved a few lines down and the dup deleted